pkg/porter/publish.go

```diff
@@ -360,15 +360,13 @@ func getNewImageNameFromBundleReference(origImg, bundleTag string, isInvocationI
 		return image.EmptyName, errors.Wrapf(err, "unable to parse bundle tag %q into domain/path components", bundleTag)
 	}
 
-	// Use the image name with the bundle location to generate a randomized tag
-	imgName := path.Join(path.Dir(bundleName.Name()), path.Base(origName.Name()))
-	nameHash := md5.Sum([]byte(imgName))
-
+	// Use the original image name with the bundle location to generate a randomized tag
+	source := path.Join(path.Dir(bundleName.Name()), path.Base(origName.Name()))
```

Comment:
Can we update this line as well so that the image includes the tag, just like you are doing above with the invocation image?

I just checked the code in cnab-to-oci and first it pushes images and then it pulls it down, _using the tag_, to determine its repository digest. I want to avoid a race condition where two people are pushing a bundle (maybe two different versions which is possible!) and then having them use the wrong repo digests because the temporary tag names we picked weren't unique.

Reply:
Do you mean something like this? 
```
source := path.Join(path.Dir(bundleName.Name()), path.Base(origName.Name())) + ":" +bundleName.Tag()

```

Reply:
Yes, thanks. 👍